contrib/scripts/run-gh-ginkgo-workflow.sh

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: Apache-2.0
+# Copyright Authors of Cilium
+
+# See also Documentation/contributing/testing/e2e.rst for reference
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+ginkgo_matrix_yaml=".github/actions/ginkgo/main-k8s-versions.yaml"
+
+check_cmd() {
+    for cmd in "$@"; do
+        if ! (command -v "$cmd" >/dev/null); then
+            return 1
+        fi
+    done
+    return 0
+}
+
+check_cmd_or_fail() {
+    for cmd in "$@"; do
+        if ! (command -v "$cmd" >/dev/null); then
+            echo "Error: $cmd not found."
+            exit 1
+        fi
+    done
+}
+
+usage() {
+    echo "Usage: $0 [options]"
+    echo "Options:"
+    echo "  -k <version>        [mandatory] Kubernetes version to use"
+    echo "  -l <version>        [optional]  Linux kernel version to use, default: derived from Kubernetes version"
+    echo "  -c <sha>            [optional]  Cilium commit to use to fetch the images, default: current commit SHA"
+    echo "  -f <focus string>   [optional]  Ginkgo focus string to use, default: 'K8s'"
+    echo "  -t <directory>      [optional]  temporary directory to store images, default: '/tmp'"
+    echo "  -w                  [optional]  assume VM image is available, just wake it up"
+    echo "  -s                  [optional]  skip straight to tests (do not set up VM or provision Kind)"
+    echo "  -h                  [optional]  show this help message"
+    exit "$1"
+}
+
+report() {
+    printf '\033[1;34m%s\033[0m\n' "$*"
+}
+
+install_helm() {
+    local HELM_VERSION=3.7.0
+
+    pushd "${tmpdir}" >/dev/null
+    wget "https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz"
+    tar -xf "helm-v${HELM_VERSION}-linux-amd64.tar.gz"
+    mv linux-amd64/helm ./helm
+    popd >/dev/null
+}
+
+install_ginkgo() {
+    go install github.com/onsi/ginkgo/ginkgo@v1.16.5
```

Comment:
renovate reference to auto-update?

Reply:
I think we use a fixed version for Ginkgo at the moment, updates seem disabled in renovate's config file because nobody has cycles to port things go Ginkgo v2+